gpu_memory_management/cupy_pybind.py

```diff
@@ -39,31 +20,41 @@ def free_gpu_memory(number_of_gpus):
     images[i] = i
 
   #cpu initial data
-  update = np.random.rand(number_of_gpus)
-  partial_update = np.zeros(number_of_elements)
+  update = random.random()
 
-  #split the input image
-  parted_images = np.array_split(images, number_of_gpus)
-  partial_update = np.array_split(partial_update, number_of_gpus)
+  print("Number of Device : ", number_of_gpus)
 
-  assert(np.shape(partial_update) == np.shape(parted_images))
+  parted_image = np.array_split(images, number_of_gpus)
 
-  print(partial_update)
+  cp.cuda.Device(0).use()
+  gpu_image_0 = cp.zeros(parted_image[0].size)
+  gpu_partial_update_0 = cp.zeros(parted_image[0].size)
 
-  # allocate memory for each GPU, the size is parted image size + update value size + parted update value size (?)
-  # need to know how to properly set the size
-  print("Number of Device : ", number_of_gpus)
-  # alloc_gpu_memory(parted_images, number_of_gpus)
+  cp.cuda.Device(1).use()
```

Comment:
Can you make the example independent of the number of GPU's please. Means, ask for the number of available GPU's in the beginning and iterate over the number.

Reply:
So number of GPU is a user input?

Reply:
No, this is not necessary. Check it in the source code, how many GPU's are available. You can restrict the number of visible CUDA GPU's via environment variable: `CUDA_VISIBLE_DEVICES=0 python example.py`.

Reply:
I already check the number of GPU in line 17 in cupy_pybind.py

Reply:
Okay, but you don't use the variable all the time. If there is only a single GPU available, this line throws an error:

https://github.com/ComputationalRadiationPhysics/student_project_python_bindings/blob/99fb7e60f171c766dfaff9c1b6fd743c97ad833d/gpu_memory_management/cupy_pybind.py#L33